keyboards/crkbd/keymaps/ajarov/keymap.c

```diff
@@ -0,0 +1,181 @@
+      XXXXXXX, ES_IEXL,   ES_AT, ES_EURO, XXXXXXX, XXXXXXX,                      XXXXXXX, XXXXXXX, XXXXXXX, ES_MORD, XXXXXXX, KC_BSPC,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      KC_LCTL, ES_FORD,  ES_NOT, ES_BSLS, ES_PIPE, ES_IQUE,                      XXXXXXX,  ES_DLR, ES_PERC, ES_AMPR, XXXXXXX,  ES_GRV,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+      KC_LSFT, XXXXXXX, XXXXXXX, ES_CCED, XXXXXXX, XXXXXXX,                      RGB_TOG, RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI, KC_RSFT,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                          KC_LGUI, _______,  KC_SPC,     KC_ENT, _______, KC_LALT
+                                      //`--------------------------'  `--------------------------'
+  )
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, 1, 2, 3);
+}
+
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  if (!is_keyboard_master()) {
+    return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+  }
+  return rotation;
+}
+
+#define L_BASE 0
+#define L_UPPER 2
+#define L_RIGHT 4
+#define L_UPPER_RIGHT 8
+
+void oled_render_layer_state(void) {
+    oled_write_P(PSTR("Layer: "), false);
+    switch (layer_state) {
+        case L_BASE:
+            oled_write_ln_P(PSTR("Default"), false);
+            break;
+        case L_UPPER:
+            oled_write_ln_P(PSTR("Upper"), false);
+            break;
+        case L_RIGHT:
+            oled_write_ln_P(PSTR("Right"), false);
+            break;
+        case L_UPPER_RIGHT:
+        case L_UPPER_RIGHT|L_UPPER:
+        case L_UPPER_RIGHT|L_RIGHT:
+        case L_UPPER_RIGHT|L_UPPER|L_RIGHT:
+            oled_write_ln_P(PSTR("Upper Right"), false);
+            break;
+    }
+}
+
+
+char keylog_str[24] = {};
+
+const char code_to_name[60] = {
+    ' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+    'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+    '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+    'R', 'E', 'B', 'T', 'S', ' ', ' ', ' ', ' ', ' ',
+    ' ', ' ', ' ', ' ', ',', '.', '-', ' ', ' ', ' '};
+
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
+           record->event.key.row, record->event.key.col,
+           keycode, name);
```

Comment:
You may want to rewrite this so that it's not using snprintf, as it can take up to 1.5k if firmware space.

https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

Reply:
I removed the keylog altogether.